opm/autodiff/StandardWell_impl.hpp

```diff
@@ -2062,6 +2064,8 @@ namespace Opm
     calculateExplicitQuantities(const Simulator& ebosSimulator,
                                 const WellState& well_state)
     {
+        updatePrimaryVariables(well_state);
+        initPrimaryVariablesEvaluation();
         computeWellConnectionPressures(ebosSimulator, well_state);
```

Comment:
I guess this is an unrelated fix. Is this just to be on the safe side or did the lack of these line cause any problems?

Reply:
Basically, when all the primary variables are not initialized, we get all oil in the wellbore. 

Q_t  and F_w and F_g are all zero, we get F_o = 1. It is what happened in the simulation. It is a fix. I did not put it in another commit, only mentioned it in the commit message. 

I will split it to two commits.

Reply:
You don't need to split it IMO, if you don't want to.

Reply:
okay. I need to revise this PR following the discussion to make the formulation easier to understand for later. We will have new Jenkins test and benchmark.